### PR TITLE
[GraphRefactor] Shortest Path

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

Transition from old ShortestPath implementation to the new refactored graph framework.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [x] **Regression tests**
  - [x] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made (required? - only internal) 
- [x] :books: documentation not required.
- [x] Backports: No backports required.

#### Related Information

- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/GORDO-41

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -133,7 +133,7 @@ set(ARANGODB_TESTS_SOURCES
   Aql/ReturnExecutorTest.cpp
   Aql/RowFetcherHelper.cpp
   Aql/ScatterExecutorTest.cpp
-  Aql/ShortestPathExecutorTest.cpp
+  #Aql/ShortestPathExecutorTest.cpp # TODO [GraphRefactor]: Re-enable / re-write again.
```
Comment (hkernbach) [inline tests/CMakeLists.txt:136]: do not forget this

Comment (hkernbach) [reply tests/CMakeLists.txt:136]: Will check this now - anyway: Can be reviewed now.

Comment (hkernbach) [reply tests/CMakeLists.txt]: Re-added and hopefully fixed.

File: arangod/Graph/Enumerators/WeightedTwoSidedEnumerator.cpp [hunk 2]
```diff
@@ -0,0 +1,714 @@
...
+  });
+}
+
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+void WeightedTwoSidedEnumerator<
+    QueueType, PathStoreType, ProviderType,
+    PathValidator>::Ball::testDepthZero(Ball& other,
+                                        CandidatesMap& candidates) {
+  // TODO: Check - is this even required?
```
Comment (hkernbach) [inline arangod/Graph/Enumerators/WeightedTwoSidedEnumerator.cpp:282]: This must be checked before we merge back. 
The min and max values are being set here: 

ShortestPathNode:413
```
  arangodb::graph::TwoSidedEnumeratorOptions enumeratorOptions{
      0, std::numeric_limits<size_t>::max(),
      arangodb::graph::PathType::Type::ShortestPath};
```

File: arangod/Graph/Enumerators/WeightedTwoSidedEnumerator.cpp [hunk 3]
```diff
@@ -0,0 +1,714 @@
...
+
+  _left.reset(source, 0);
+  _right.reset(target, 0);
+  _resultPath.clear();
+
+  // Special depth == 0 case
+  if (_options.getMinDepth() == 0 && source == target) {
+    _left.testDepthZero(_right, _candidates);
+  }
+}
```
Comment (hkernbach) [inline arangod/Graph/Enumerators/WeightedTwoSidedEnumerator.cpp:423]: Originally transferred from the "TwoSidedEnumerator". Info:

This must be checked before we merge back. 
The min and max values are being set here: 

ShortestPathNode:413
```
  arangodb::graph::TwoSidedEnumeratorOptions enumeratorOptions{
      0, std::numeric_limits<size_t>::max(),
      arangodb::graph::PathType::Type::ShortestPath};
```

Comment (hkernbach) [reply arangod/Graph/Enumerators/WeightedTwoSidedEnumerator.cpp:423]: Currently the implementation does not do anything (blank body)

**Review by markuspf — APPROVED** [review]
LGTM

**Pull request merged**